Plugins/Public/cloak_plugin/Cloak.cpp

```diff
@@ -291,51 +309,40 @@ void SetState(uint iClientID, uint iShipID, int iNewState)
 	}
 }
 
-bool removeSingleFuel(uint iClientID, CLOAK_INFO &info)
-{
-	for (list<EquipDesc>::iterator item = Players[iClientID].equipDescList.equip.begin(); item != Players[iClientID].equipDescList.equip.end(); item++)
-	{
-		uint fuel_usage = info.arch.mapFuelToUsage[item->iArchID];
-		if (item->iCount >= fuel_usage)
-		{
-			pub::Player::RemoveCargo(iClientID, item->sID, fuel_usage);
-			info.singleCloakConsumed = true;
-			return true;
-		}
-	}
-
-	return false;
-
-}
-
 // Returns false if the ship has no fuel to operate its cloaking device.
-static bool ProcessFuel(uint iClientID, CLOAK_INFO &info)
+static bool ProcessFuel(uint iClientID, CLOAK_INFO &info, uint iShipID)
 {
 	if (info.bAdmin)
 		return true;
 
-	if (!info.singleCloakConsumed && info.arch.singleUseCloak) {
-		return removeSingleFuel(iClientID, info);
-	}
-
-	else if (info.singleCloakConsumed)
-	{
+	if(setJumpingClients.find(iClientID) != setJumpingClients.end())
 		return true;
-	}
 
 	for (list<EquipDesc>::iterator item = Players[iClientID].equipDescList.equip.begin(); item != Players[iClientID].equipDescList.equip.end(); item++)
 	{
 		if (info.arch.mapFuelToUsage.find(item->iArchID) != info.arch.mapFuelToUsage.end())
 		{
-			uint fuel_usage = info.arch.mapFuelToUsage[item->iArchID];
-			if (item->iCount >= fuel_usage)
+			const auto& fuelUsage = info.arch.mapFuelToUsage[item->iArchID];
+			float currFuelUsage = fuelUsage.usageStatic;
+			if (fuelUsage.usageLinear != 0.0f || fuelUsage.usageSquare != 0.0f) {
+				Vector dir1;
+				Vector dir2;
+				pub::SpaceObj::GetMotion(iShipID, dir1, dir2);
+				float vecLength = sqrtf(dir1.x * dir1.x + dir1.y * dir1.y + dir1.z * dir1.z);
+				
+				currFuelUsage += fuelUsage.usageLinear * vecLength;
+				currFuelUsage += fuelUsage.usageSquare * vecLength * vecLength;
+			}
+			uint totalFuelUsage = static_cast<uint>(max(currFuelUsage, 0.0f));
+			if (item->iCount >= totalFuelUsage)
 			{
-				pub::Player::RemoveCargo(iClientID, item->sID, fuel_usage);
+				pub::Player::RemoveCargo(iClientID, item->sID, totalFuelUsage);
 				return true;
 			}
+			if(info.arch.mapFuelToUsage.size() == 1)

```

Comment:
is this some sort of weird optimisation for specific configs?

Reply:
If we're using a single fuel commodity and we just finished processing it, there's no reason to continue looking, so we break the loop early.

Reply:
ok I guess, stops us from having to iterate through the rest of the player's equipDescList